pcdscalc/tests/test_be_lens_calcs.py

```diff
@@ -356,21 +351,40 @@ def test_calc_lens_set():
                              [1, 0, 0, 0, 0]])
 
     expected_effrads = np.array([0.001, 0.0005, 0.0003, 0.0002, 0.0001])
-    expected_sizes = np.array([0.00047886, 0.00045743, 0.000429,
-                              0.00039348, 0.00028694])
-    expected_foclens = np.array([93.87107082, 46.93553541, 28.16132124,
-                                18.77421416,  9.38710708])
+    expected_sizes = np.array([0.00047885, 0.0004574, 0.00042894,
+                              0.0003934, 0.00028678])
+    expected_foclens = np.array([93.80033687, 46.90016844, 28.14010106,
+                                 18.76006737, 9.38003369])
 
     res = be_lens_calcs.calc_lens_set(energy=8, size_fwhm=2, distance=4,
                                       n_max=1, max_each=2,
                                       lens_radii=[100e-6, 200e-6, 300e-6,
                                                   500e-6, 1000e-6],
                                       fwhm_unfocused=0.0005, eff_rad0=None)
-    sets = res[0]
-    effrads = res[1]
-    sizes = res[2]
-    foclens = res[3]
-    assert sets.all() == expected_sets.all()
-    assert effrads.all() == expected_effrads.all()
-    assert sizes.all() == expected_sizes.all()
-    assert foclens.all() == expected_foclens.all()
+
+    sets, effrads, sizes, foclens = res
+
+    assert np.allclose(sets, expected_sets)
+    assert np.allclose(effrads, expected_effrads)
+    assert np.allclose(sizes, expected_sizes)
+    assert np.allclose(foclens, expected_foclens)
+
+
+@pytest.mark.parametrize('radius, fwhm, energy, expected', [
+                         pytest.param(2, 200e-6, 8, 0.9937771825941067),
+                         pytest.param(4, 500e-6, 9, 0.9953931501494162),
+                         pytest.param(6, 300e-6, 10, 0.9964134300156009)
+                         ])
+def test_lens_transmission(radius, fwhm, energy, expected):
+    res = be_lens_calcs.lens_transmission(r=radius, fwhm=fwhm, energy=energy)
+    logger.debug('Expected: %s Received: %s', expected, res)
+    assert np.isclose(expected, res)
+
+
+# with good_sets = [False, True, True, False, False, True, False, False, False]
+def test_plan_set():
```

Comment:
It is hard to make the `plan_set` generate some `True` for `good_sets` to be able to test the results....  I've been testing it with dummy `good_sets` against the old function and I've been getting the same results, but obviously I have to find some good parameters to test this

Reply:
Seems good enough to me. The values provided to that function don't seem to change the control flow, meaning testing arbitrary-ish values against the old implementation is likely sufficient.

Reply:
I agree with Ken, this is good